ksql-engine/src/main/java/io/confluent/ksql/KsqlEngine.java

```diff
@@ -303,7 +316,7 @@ public QueryMetadata getQueryExecutionPlan(final Query query, final KsqlConfig k
       // Parse and AST creation
       final KsqlParser ksqlParser = new KsqlParser();
 
-      final List<PreparedStatement> statements = ksqlParser.buildAst(
+      return ksqlParser.buildAst(
           queriesString,
           tempMetaStoreForParser,
           stmt -> buildSingleQueryAst(
```

Comment:
Because if you're parsing multiple statements its possible that latter statements rely on what `buildSingleQueryAst` does with a former statement, e.g. a CSAS relying on a previous CS.

I agree though - it's on my radar.

Reply:
Ah right. Thanks!